eZ/Bundle/EzPublishCoreBundle/Converter/RepositoryParamConverter.php

```diff
@@ -0,0 +1,108 @@
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+class RepositoryParamConverter implements ParamConverterInterface
+{
+    const LOCATION_ID = 'locationId';
+    const CONTENT_ID = 'contentId';
+
+    const LOCATION_CLASS = 'eZ\Publish\API\Repository\Values\Content\Location';
+    const CONTENT_CLASS = 'eZ\Publish\API\Repository\Values\Content\Content';
+
+    private $supportedClasses;
+
+    /** @var  \eZ\Publish\API\Repository\LocationService */
+    private $locationService;
+
+    /** @var \eZ\Publish\API\Repository\ContentService */
+    private $contentService;
+
+    public function __construct( LocationService $locationService, ContentService $contentService )
+    {
+        $this->locationService = $locationService;
+        $this->contentService = $contentService;
+        $this->supportedClasses = [ self::LOCATION_CLASS, self::CONTENT_CLASS ];
+    }
+
+    public function apply( Request $request, ParamConverter $configuration )
+    {
+        try
+        {
+            if ( $configuration->getClass() === self::LOCATION_CLASS )
+            {
+                if ( !$request->attributes->has( self::LOCATION_ID ) )
+                {
+                    return false;
+                }
+
+                $value = $request->attributes->get( self::LOCATION_ID );
+                if ( !$value && $configuration->isOptional() )
+                {
+                    return false;
+                }
+
+                $request->attributes->set( $configuration->getName(), $this->locationService->loadLocation( $value ) );
+                return true;
+            }
+            else if ( $configuration->getClass() === self::CONTENT_CLASS )
+            {
+                if ( !$request->attributes->has( self::CONTENT_ID ) )
+                {
+                    return false;
+                }
+
+                $value = $request->attributes->get( self::CONTENT_ID );
+                if ( !$value && $configuration->isOptional() )
+                {
+                    return false;
+                }
+
+                $request->attributes->set( $configuration->getName(), $this->contentService->loadContent( $value ) );
+                return true;
+            }
+        }
+        catch ( NotFoundException $e )
+        {
+            throw new NotFoundHttpException( 'Requested values not found', $e );
+        }
+        catch ( UnauthorizedException $e )
+        {
+            throw new AccessDeniedHttpException( 'Access to values denied', $e );
+        }
+
+        return false;
+    }
+
+    public function supports( ParamConverter $configuration )
+    {
+        if ( $configuration->getClass() === null )
+        {
+            return false;
+        }
+
+        if ( in_array( $configuration->getClass(), $this->supportedClasses ) )
+        {
+            return true;
+        }
+
+        return false;
```

Comment:
Implementation of this method is too complex.

Suggestion: 

```php
public function supports( ParamConverter $configuration )
{
    if ( !isset( $this->supportedClasses[(string)$configuration->getClass()] ) )
    {
        return false;
    }

    return true;
}
```

And in Constructor, change `$this->supportedClasses` assignment:

```php
$this->supportedClasses = [self::LOCATION_CLASS => true, self::CONTENT_CLASS => true];
```

This avoids call in `in_array()`, which is a lot more costy than `isset()`.